### PR TITLE
Support bread crumb drag

### LINKED ISSUE #182685: Allow dragging and dropping file entries in the breadcrumbs
Type: <b>Bug</b>

**Feature request**
Suport drag and dropping of file entries in the breadcrumbs

This should integrate with VS Code's drag and drop into editor so that you can drag a breadcrumb element into a file to insert a path

VS Code version: Code - OSS Dev 1.79.0 (Commit unknown, Date unknown)
OS version: Darwin x64 22.4.0
Modes:

### PR DESCRIPTION
**Pull request by Kalmaegi**
Fix #182685

### CODE REVIEW

**Comment by weartist** [comment]
@mjbvz hi I found two question:
1. When using drag breadcrumb, I found that if there is a file in the current breadcrumbs, this file is the currently active file, so dragging and turning on this function is not necessary
2. When I drag and drop outside the edit area (e.g. file list, tab), what do we expect to happen

File: src/vs/base/browser/ui/breadcrumbs/breadcrumbsWidget.ts [hunk 1]
```diff
@@ -3,17 +3,21 @@
...
 import { commonPrefixLength } from 'vs/base/common/arrays';
-import { ThemeIcon } from 'vs/base/common/themables';
 import { Emitter, Event } from 'vs/base/common/event';
-import { DisposableStore, dispose, IDisposable } from 'vs/base/common/lifecycle';
+import { DisposableStore, IDisposable, dispose } from 'vs/base/common/lifecycle';
 import { ScrollbarVisibility } from 'vs/base/common/scrollable';
-import 'vs/css!./breadcrumbsWidget';
+import { ThemeIcon } from 'vs/base/common/themables';
+import { FileKind } from 'vs/platform/files/common/files';
+import { FileElement, OutlineElement2 } from 'vs/workbench/browser/parts/editor/breadcrumbsModel';
```
Comment (jrieken) [inline src/vs/base/browser/ui/breadcrumbs/breadcrumbsWidget.ts:17]: This introduces layer breakers. Code in `/base` cannot access code from `/workbench` or `/platform`. See https://github.com/microsoft/vscode/wiki/Source-Code-Organization#layers for more details

File: src/vs/workbench/browser/parts/editor/breadcrumbsControl.ts [hunk 2]
```diff
@@ -4,49 +4,49 @@
...
-import { ACTIVE_GROUP, ACTIVE_GROUP_TYPE, IEditorService, SIDE_GROUP, SIDE_GROUP_TYPE } from 'vs/workbench/services/editor/common/editorService';
-import { IEditorGroupsService } from 'vs/workbench/services/editor/common/editorGroupsService';
 import { IEditorGroupView } from 'vs/workbench/browser/parts/editor/editor';
-import { PixelRatio } from 'vs/base/browser/browser';
-import { ILabelService } from 'vs/platform/label/common/label';
-import { Categories } from 'vs/platform/action/common/actionCommonCategories';
-import { ITreeNode } from 'vs/base/browser/ui/tree/tree';
+import { EditorResourceAccessor, IEditorPartOptions, SideBySideEditor } from 'vs/workbench/common/editor';
+import { IEditorGroupsService } from 'vs/workbench/services/editor/common/editorGroupsService';
+import { ACTIVE_GROUP, ACTIVE_GROUP_TYPE, IEditorService, SIDE_GROUP, SIDE_GROUP_TYPE } from 'vs/workbench/services/editor/common/editorService';
```
Comment (jrieken) [inline src/vs/workbench/browser/parts/editor/breadcrumbsControl.ts:40]: Don't make cosmetic changes like resorting imports

**Review by jrieken — CHANGES_REQUESTED** [review]

File: src/vs/workbench/browser/parts/editor/breadcrumbsControl.ts (hunk 2 above)
Comment (weartist) [reply src/vs/workbench/browser/parts/editor/breadcrumbsControl.ts:40]: oh sorry,  i will resubmit a PR

File: src/vs/base/browser/ui/breadcrumbs/breadcrumbsWidget.ts (hunk 1 above)
Comment (weartist) [reply src/vs/base/browser/ui/breadcrumbs/breadcrumbsWidget.ts:17]: thanks！I will correct here

**Pull request closed without merging**